db/system_distributed_keyspace.cc

```diff
@@ -579,7 +579,7 @@ system_distributed_keyspace::read_cdc_generation(utils::UUID id) {
     // Paranoic sanity check. Partial reads should not happen since generations should be retrieved only after they
     // were written successfully with CL=ALL. But nobody uses EverywhereStrategy tables so they weren't ever properly
     // tested, so just in case...
-    if (entries.size() != num_ranges) {
+    if (std::cmp_not_equal(entries.size(), num_ranges)) {
```

Comment:
Just replace `auto num_ranges = 0` with `size_t num_ranges = 0`.

Reply:
what about the 

```c++
num_ranges = row.get_as<int32_t>("num_ranges");
```

couple lines above. shall we error out at seeing a negative "num_ranges"?  i also wanted to have a simpler fix, but this change would introduce more risk.

Reply:
The only reason `num_ranges` is signed is because C* (being Java) had no unsigned integer types.

Reply:
i see. that's a shame, even LLVM has unsigned integer..

Reply:
i will add a follow-up change to implement your suggestion.

Reply:
#14754